app/services/arcgis_api/mock/geocoder.rb

```diff
@@ -0,0 +1,51 @@
+module ArcgisApi
+  module Mock
+    class Geocoder < ArcgisApi::Geocoder
+      def initialize
+        super do |conn|
+          conn.adapter :test do |stub|
```

Comment:
another idea, what if we didn't do this in the initializer, but added an optional block to the `faraday` method and override that?

```
def faraday
  super do |conn|
  # ...
  end
end
end
```